db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -32,6 +32,7 @@ abstract class GrpcClient {
     protected Exception lastException;
     protected UUID currentChannelId;
     protected Optional<ServerInfo> serverInfo = Optional.empty();
+    private volatile boolean serverInfoLoaded = false;
```

Comment:
Why it's needed to know if we have already loaded the server info or not?

Reply:
this is to detect if we've already connected to at least one node and fetched the server info in order to make sure that when getServerVersion() is called, we either return a correct value (which may be slightly stale if the client connects to another node) or throw an exception if it's not populated yet

Reply:
See: https://github.com/EventStore/EventStoreDB-Client-Java/pull/165#discussion_r930520577